llm_toolkit/crash_triage.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Triaging the crash with LLM."""
+
+import argparse
+import os
+import sys
+
+from experiment import benchmark as benchmarklib
+from llm_toolkit import models
+from llm_toolkit import output_parser as parser
+from llm_toolkit import prompt_builder
+
+
+def parse_args():
+  """Parses command line arguments."""
+  argparser = argparse.ArgumentParser(description='Triage the crash with LLM.')
+  argparser.add_argument(
+      '-t',
+      '--target-dir',
+      type=str,
+      default='./targets',
+      help='The directory to store all fuzz targets to be triaged.')
+  argparser.add_argument(
+      '-o',
+      '--intermediate-output-dir',
+      type=str,
+      default='./output',
+      help=('The directory to store all intermediate output files (LLM prompt, '
+            'rawoutput).'))
+  argparser.add_argument('-p',
+                         '--project',
+                         type=str,
+                         required=True,
+                         help='The project name.')
+  argparser.add_argument('-i',
+                         '--crash-info',
+                         type=str,
+                         required=True,
+                         help='The directory to store all crash information.')
+  argparser.add_argument(
+      '-c',
+      '--code',
+      type=str,
+      required=True,
+      help='The directory to store all crash-related code (project code).')
+
+  args = argparser.parse_args()
+  if args.target_dir and os.listdir(args.target_dir):
+    assert os.path.isdir(
+        args.target_dir
+    ), f'--target-dir must take an existing directory: {args.target_dir}.'
+    assert os.listdir(
+        args.target_dir
+    ), f'--target-dir must take a non-empty directory: {args.target_dir}.'
+
+  os.makedirs(args.intermediate_output_dir, exist_ok=True)
+
+  return args
+
+
+def triage_all_crashes():
+  """Reads crash information and relevant code, applies triage, \
+        and saves the result."""
+  # TODO(fdt622): Finish this.
+
+
+# ========================= LLM Triage ========================= #
+
+
+def llm_triage(
```

Comment:
Thanks for following the same structure as `llm_fix`, this simplifies readability and maintainability.
Also, we might need to convert this to `llm_fix` later, if we want to test asking LLM to fix code based on its own triage response : )

Reply:
Okay, I'll convert it when we need to. For now I keep it as llm_triage.